lib/modules/actions.js

```diff
@@ -54,7 +56,7 @@ export const reroutePage = () => async (dispatch, getState) => {
       queryParams: currentPage.queryParams,
       hashParams: currentPage.hashParams,
       bodyParams: {},
-      referrer: currentPage.referrer,
+      referrer: currentPage.referrer || NO_REFERRER,
```

Comment:
so in mweb if I say referrer is what platform sends as long as history length is less than 3 then it breaks if/when this changes. Its not necessarily a "bug", more like a design compromise due to how the shell works.